client/app/dbaas/metrics/dashboard/metrics-dashboard.controller.js

```diff
@@ -55,6 +58,16 @@
                 .finally(() => {
                     this.loading.consumption = false;
                 });
+            this.MetricService.getServiceInfos(this.serviceName)
+                .then(info => {
+                    this.plan.autorenew = moment(info.data.expiration).format("LL");
```

Comment:
I'm not asking you to refactor this 

You can now use something like 

this.instances = this.ControllerHelper.request.getArray(or Hash)Loader({
            loaderFunction: () => this.CloudDbProjectService.getInstances(this.projectId)
        });

thjen call this.instances.load();

you'll have access to =>
this.instances.loading to know if it's still loading or not
this.instances.hasErrors to see if something wrong happened (not useful most of the time)
this.instances.data... to use your data :) (initialized to null... remains null until loading is done)

Reply:
Okay, noticed, will be done with next release of Metrics.